be/src/io/fs/file_system.h

```diff
@@ -45,6 +45,7 @@ namespace io {
         if (!_s) {                            \
             LOG(WARNING) << _s;               \
         }                                     \
+        _s.reset_stack();                     \
```

Comment:
If you do not want the stack any more, then you could create a new status object and not add stack strace.
For example,  Error<error_code,false>